sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -104,7 +105,7 @@ case class HiveTableScanExec(
     hadoopConf)
 
   private def castFromString(value: String, dataType: DataType) = {
-    Cast(Literal(value), dataType).eval(null)
+    Cast(Literal(value), dataType, Option(SQLConf.get.sessionLocalTimeZone)).eval(null)
```

Comment:
BTW, is it elegant to initialize a `CastSupport` (`DataSourceAnalysis` rule or `DataSourceStrategy`) here, in which we still need to pass `SQLConf`?